### PR TITLE
validations and model specs

### PR DESCRIPTION
**Pull request by sam2if**
### Project requirements
- Add the following validations:
  - For the User model:
    - Name must not be blank.
    - PostsCounter must be an integer greater than or equal to zero.
  - For the Post model:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than or equal to zero.
    - LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.

### CODE REVIEW

**Review by Abenezer-Tilahun — CHANGES_REQUESTED** [review]
Hi @sam2if 

Great job for implemented  the some of the Project Requirement  , but there are some changes that need to be implemented in order to proceed to the next project 👍

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README
✔️ Good Commit Messages
✔️ Good job for Implemented Blog App -validations and model specs..
✔️ All Test Cases are Passing. 👍.

image


almost

## Required Changes ♻️
_Check the comments below the review._

- [x] As stated in the project requirement Kindly make sur that you Added **unit specs for all of your models' methods and validations.** But you only add unit specs for User and post model but Kindly I recommend you to **add unit tests for Comment, and Like model**.

- [x] Kindly I recommend you to include how to run the test locally on machine on the Get Stared section of the Readme File So kindly Update it.
image


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/comment_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      # Call the method to update the comments_counter
+      comment.update_post_comments_counter
+
+      # Reload the post from the database to get the updated comments_counter
+      post.reload
+
+      expect(post.comments_counter).to eq(initial_comments_count + 1)
+    end
+  end
+end
```
Comment (prolajumokeoni) [inline spec/models/comment_spec.rb]: Screenshot_235

- [ ] Kindly make sure that all tests added are passing only tests in your user and post spec files are passing, the tests added to the comment and  like spec files are failing please make sure alll your tests are passing 👍🏿

File: spec/models/like_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      # Call the method to update the likes_counter
+      like.update_post_likes_counter
+
+      # Reload the post from the database to get the updated likes_counter
+      post.reload
+
+      expect(post.likes_counter).to eq(initial_likes_count + 1)
+    end
+  end
+end
```
Comment (prolajumokeoni) [inline spec/models/like_spec.rb]: - [ ] Kindly make sure that all tests added are passing only tests in your user and post spec files are passing, the tests added to the comment and  like spec files are failing please make sure alll your tests are passing, see screenshot added above  👍🏿

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @sam2if 
Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Highlights
- Created spec files for like and comment ✔️ 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Davidosky007 — APPROVED** [review]
Hi @sam2if 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* All tests passed 👍 
* No linter errors 👍 
* All validations created properly 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**